accepted/future-releases/async-star-behavior/feature-specification.md

```diff
@@ -0,0 +1,108 @@
+If the subscription is canceled, delivering succeeds and does nothing,
+if the subscription is a paused,
+delivery succeeds when the event is accepted and buffered.
+Otherwise, deliver is successful after the subscription's event listener
+has been invoked with the event object.
+
+After this has happened, the subscription is checked for being
+canceled or paused.
+If paused, the function is blocked at the `yield` statement until
+the subscription is resumed or canceled.
+In this case the `yield` is an asynchronous operation (it does not complete
+synchronously, but waits for an external event, the resume,
+before it continues).
+If canceled, including if the cancel happens during a pause,
+the `yield` statement acts like a `return;` statement.
+
+A `yield* e;` statement listens on the stream that `e` evaluates to
+and forwards all events to this function's subscription.
+If the subscription is paused, the pause is forwarded to the yielded stream
+If the subscription is canceled, the cancel is forwarded to the yielded stream,
+then the `yield*` statement waits for any cancellation future, and finally
+it acts like a `return;` statement.
+If the yielded stream completes, the yield statement completes normally.
+A `yield*` is *always* an asynchronous operation.
+
+In an asynchronous function, an `await for (var event in stream) ...` loop
+first listens on the iterated stream, then for each data event, it executes the
+body. If the body performs any asynchronous operation (that is,
+it does not complete synchronously because it executes any `await`,
+`wait for` or `yield*` operation, or it blocks at a `yield`), then
+the stream subscription must be paused. It is resumed again when the
+body completes normally. If the loop body breaks the loop (by any means,
+including throwing or breaking), or if the iterated stream produces an error,
+then the loop is broken. Then the subscription is canceled and the cancellation
+future is awaited, and then the loop completes in the same way as the body
+or by throwing the produced error and its accompanying stack trace.
+
+Notice that there is no requirement that an `async*` implementation must call
+the subscription event handler *synchronously*, but if not, then it must
+block at the `yield` until the event has been delivered. Since it's possible
+to deliver the event synchronously, it's likely that that will be the
+implementation, and it's possible that performance may improve due to this.
+
+### Consequences
+Implementations currently do not block at a `yield` when the delivery of
+the event causes a pause. They simply does not allow a `yield` statement
+to act asynchronously. They can *cancel* at a yield if the cancel happened
+prior to the `yield`, and can easily be made to respect a cancel happening
+during the `yield` event delivery, but they only check for pause *before*
+delivering the event, and it requires a rewrite of the Kernel transformer
+to change this behavior.
```

Comment:
It would be fantastic if we could add an example section that shows how these work together!

Reply:
Example addded.